application/controllers/ServicesController.php

```diff
@@ -198,6 +202,105 @@ public function searchEditorAction()
+            'host_name' => 'host.display_name',
+            'host_display_name' => 'host.name',
+            'service_name' => 'service.name',
+            'service_display_name' => 'service.display_name',
+            'service_handled' => 'service.state.is_handled',
+            'service_output' => 'service.state.output',
+            'service_state' => 'service.state.soft_state'
+        ];
+
+        $pivotFilter = (bool) $this->params->get('problems', false) ?
+            Filter::equal('service.state.is_problem', 'y') : null;
+
+        $problemToggle = $this->createProblemToggle();
+        $this->addControl($problemToggle);
+
+        $sortControl = $this->createSortControl($query, [
+            'host.name'     => $this->translate('Host Name'),
+            'service.name'  => $this->translate('Service Name')
+        ]);
+        $flipped = $this->params->shift('flipped', false);
+
+        if ($flipped) {
+            $xAxisCol = 'host_name';
+            $yAxisCol = 'service_name';
+        } else {
+            $xAxisCol = 'service_name';
+            $yAxisCol = 'host_name';
+        }
+
+        $this->params->shift('page');
+        $searchBar = $this->createSearchBar($query, [
+            LimitControl::DEFAULT_LIMIT_PARAM,
+            $sortControl->getSortParam(),
+            'flipped',
+            'page',
+            'problems'
+        ]);
+
+        if ($searchBar->hasBeenSent() && ! $searchBar->isValid()) {
+            if ($searchBar->hasBeenSubmitted()) {
+                $filter = $this->getFilter();
+            } else {
+                $this->addControl($searchBar);
+                $this->sendMultipartUpdate();
+                return;
+            }
+        } else {
+            $filter = $searchBar->getFilter();
+        }
+
+        $continueWith = $this->createContinueWith(
+            Url::fromPath('icingadb/services/grid')->addParams($this->getAllParams()),
+            $searchBar
+        );
+
+        $this->filter($query, $filter);
+
+        $this->addControl($sortControl);
+        $this->addControl($searchBar);
+
+        $pivot = (new PivotTable($query, $xAxisCol, $yAxisCol, $gridcols))
+            ->setXAxisFilter($pivotFilter)
+            ->setYAxisFilter($pivotFilter ? clone $pivotFilter : null)
+            ->setXAxisHeader($xAxisCol)
+            ->setYAxisHeader($yAxisCol);
+
+        $this->view->horizontalPaginator = $pivot->paginateXAxis();
+        $this->view->verticalPaginator = $pivot->paginateYAxis();
+        list($pivotData, $pivotHeader) = $pivot->toArray();
+        $this->view->pivotData = $pivotData;
+        $this->view->pivotHeader = $pivotHeader;
+
+        $this->view->controls = $this->controls;
+
+        if ($flipped) {
+            $this->render('grid-flipped');
+        }
+
+        if (! $searchBar->hasBeenSubmitted() && $searchBar->hasBeenSent()) {
+            $this->sendMultipartUpdate($continueWith);
+        }
+
+        $this->setAutorefreshInterval(10);
```

Comment:
Please increase this to 30 seconds. The grid may become very large quickly, refreshing every 10th second is too often then.